discovery/marathon/marathon_test.go

```diff
@@ -153,40 +133,27 @@ func TestMarathonSDRemoveApp(t *testing.T) {
 	defer refreshMetrics.Unregister()
 
 	md, err := NewDiscovery(cfg, nil, metrics)
-	if err != nil {
-		t.Fatalf("%s", err)
-	}
+	require.NoError(t, err)
 
 	md.appsClient = func(_ context.Context, _ *http.Client, _ string) (*appList, error) {
 		return marathonTestAppList(marathonValidLabel, 1), nil
 	}
 	tgs, err := md.refresh(context.Background())
-	if err != nil {
-		t.Fatalf("Got error on first update: %s", err)
-	}
-	if len(tgs) != 1 {
-		t.Fatal("Expected 1 targetgroup, got", len(tgs))
-	}
+	require.NoError(t, err, "Got error on first update.")
+	require.Len(t, tgs, 1, "Expected 1 targetgroup.")
 	tg1 := tgs[0]
 
 	md.appsClient = func(_ context.Context, _ *http.Client, _ string) (*appList, error) {
 		return marathonTestAppList(marathonValidLabel, 0), nil
 	}
 	tgs, err = md.refresh(context.Background())
-	if err != nil {
-		t.Fatalf("Got error on second update: %s", err)
-	}
-	if len(tgs) != 1 {
-		t.Fatal("Expected 1 targetgroup, got", len(tgs))
-	}
+	require.NoError(t, err, "Got error on second update.")
+	require.Len(t, tgs, 1, "Expected 1 targetgroup.")
+
 	tg2 := tgs[0]
 
-	if tg2.Source != tg1.Source {
-		if len(tg2.Targets) > 0 {
-			t.Errorf("Got a non-empty target set: %s", tg2.Targets)
-		}
-		t.Fatalf("Source is different: %s != %s", tg1.Source, tg2.Source)
-	}
+	require.Equal(t, tg1.Source, tg2.Source, "Source is different.")
+	require.NotEmpty(t, tg2.Targets, "Got a non-empty target set.")
```

Comment:
This would somewhat chance the functionality of the test because it would fail immediately instead of checking for the non emtpy target set first. I don't know if we care about that, but if we don't then we should remove the non-empty target set error because then that should never go off.

Reply:
I don't understand the meaning of the test either.  I will swap the lines round; previously we would report both errors and now we will report just one, but I guess someone working on it can cope.